src/utils/tenant.ts

```diff
@@ -0,0 +1,79 @@
+import { runMigrationsOnTenant } from './migrate'
+import { pool } from './multitenant-db'
+
+interface TenantConfig {
+  anonKey: string
+  databaseUrl: string
+  jwtSecret: string
+  serviceKey: string
+}
+
+const tenantConfigCache: {
+  [tenantId: string]: TenantConfig
+} = {}
+
+export async function cacheTenantConfigAndRunMigrations(
+  tenantId: string,
+  config: TenantConfig
+): Promise<void> {
+  await runMigrationsOnTenant(config.databaseUrl)
+  tenantConfigCache[tenantId] = config
+}
+
+export function deleteTenantConfig(tenantId: string): void {
+  delete tenantConfigCache[tenantId]
+}
+
+export async function cacheTenantConfigsFromDbAndRunMigrations(): Promise<void> {
+  const result = await pool.query(
+    `
+    SELECT
+      id,
+      config
+    FROM
+      storage.tenants
+    `
+  )
+  for (const tenant of result.rows) {
+    const { id, config } = tenant
+    await cacheTenantConfigAndRunMigrations(id, config)
```

Comment:
Should we do this in batches of 100 or so? Connecting to 1000s of dbs individually and running migrations sequentially will take a considerable amount of time

Reply:
Done.